src/html-serializer.js

```diff
@@ -0,0 +1,188 @@
+   */
+  constructor({ imageCopyrightAttribute } = {}) {
+    this.imageCopyrightAttribute = imageCopyrightAttribute || 'data-copyright'
+  }
+
+  /**
+   * Serialize node tree into array of HTML strings.
+   *
+   * @param {NodeTree} nodeTree
+   * @returns {string[]} serialized node tree
+   */
+  serializeNodeTree(nodeTree) {
+    const step = node => {
+      const serializedChildren = node.children.reduce(
+        (result, node) => result.concat([step(node)]),
+        []
+      )
+
+      return this.serialize(node, serializedChildren)
+    }
+
+    return nodeTree.children.map(parentNode => {
+      return step(parentNode)
+    })
+  }
+
+  /**
+   * Create <tag> with children inside.
+   *
+   * @param {string} tag - Outer HTML element tag
+   * @param {string[]} children - Children concatenated and used as inner HTML
+   * @returns {string} serialized HTML string (Tag with children)
+   */
+  serializeStandardTag(tag, children) {
+    return `<${tag}>${children.join('')}</${tag}>`
+  }
+
+  /**
+   * Create <img> wrapped in <p>.
+   * Image is additionally wrapped in <a> tag, if linkTo is provided on the element.
+   *
+   * @param element - Prismic image element
+   * @returns {string} serialized HTML string (Image)
+   */
+  serializeImage(element) {
+    const src = escapeHtml(element.url)
+    const alt = escapeHtml(element.alt || '')
+    const copyright = escapeHtml(element.copyright || '')
+
+    const img = `<img src="${src}" alt="${alt}" ${this.imageCopyrightAttribute}="${copyright}">`
+
+    let content = element.linkTo
+      ? HtmlSerializer.serializeHyperlink(element.linkTo, img)
+      : img
+
+    return `<p>${content}</p>`
+  }
+
+  /**
+   * Create <div> containing Prismic embed HTML.
+   *
+   * @param element - Prismic embed element
+   * @returns {string} serialized HTML string (Div with e.g. iframe inside)
+   */
+  serializeEmbed(element) {
+    const dataOembed = escapeHtml(element.oembed.embed_url)
+    const dataOembedType = escapeHtml(element.oembed.type)
+    const dataOembedProvider = escapeHtml(element.oembed.provider_name)
+
+    return `
+        <div data-oembed="${dataOembed}"
+          data-oembed-type="${dataOembedType}"
+          data-oembed-provider="${dataOembedProvider}"
+        >
+          ${element.oembed.html}
+        </div>
+      `
+  }
+
+  /**
+   * Create <a> with content inside.
+   *
+   * @param link - Prismic link with optional target and url
+   * @param {string} [content=] - Content to be used as inner HTML
+   * @returns {string} serialized HTML string (Link)
+   */
+  serializeHyperlink(link, content = '') {
+    const href = escapeHtml(resolveUrl(link))
+    const escapedTarget = escapeHtml(link.target || '')
+    const target = escapedTarget
+      ? ` target="${escapedTarget}" rel="noopener" `
+      : ' '
+
+    return `<a${target}href="${href}">${content}</a>`
```

Comment:
Would
```
const targetSegment = escapedTarget
      ? `target="${escapedTarget}" rel="noopener"`
      : ''

    return `<a ${targetSegment} href="${href}">${content}</a>`
```
be more in control?

Reply:
What do you mean by more in control? It is like this in the prismic-dom lib, but when not providing a target it leaves the a tag with two whitespaces, which seems a bit random to me (let a test fail and was hard to see).

Reply:
I only mean the ' '